### PR TITLE
feat: sp service uses new error and error code

### PR DESCRIPTION
**Pull request by sysvm**
### Description

SP service uses new error and error code.

### Rationale

N/A

### Example

N/A

### Changes

N/A

### CODE REVIEW

File: store/sqldb/object_integrity.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ func (s *SpDBImpl) GetObjectIntegrity(objectID uint64) (*IntegrityMeta, error) {
 		return nil, result.Error
 	}
 	if result.Error != nil {
-		return nil, fmt.Errorf("failed to query integrity meta record: %s", result.Error)
+		return nil, fmt.Errorf("failed to query  record in integrity meta table: %s", result.Error)
```
Comment (will-2012) [inline store/sqldb/object_integrity.go:23]: `query  record` has extra spaces.

Comment (sysvm) [reply store/sqldb/object_integrity.go:23]: fixed

File: model/errors/error_code.go [hunk 2]
```diff
@@ -10,3 +10,41 @@ const (
 	// ErrorCodeInternalError defines internal error code
 	ErrorCodeInternalError = 50001
 )
+
+const (
+	// parse error
+	ParseStrToIntErrCode         = 15000
+	ConvertStrToByteSliceErrCode = 15001
+	// sp database error, from 20000 to 24999
+	QueryInSPInfoTableErrCode         = 20000
```
[the remark this answers is not in the document]
Comment (sysvm) [reply model/errors/error_code.go:19]: this is a query operation error code

[the remark this answers is not in the document]
Comment (sysvm) [reply model/errors/error_code.go:19]: now this code is deleted.

File: pkg/errors/types/error.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}
+}
+
+// Code gets error code from an error
+func Code(e error) int {
+	if e == nil {
+		return 0 // 0 represents success
+	}
+	err, ok := e.(*ServiceError)
+	if !ok && errors.As(e, &err) {
```
[the remark this answers is not in the document]
Comment (sysvm) [reply pkg/errors/types/error.go:58]: fixed

File: util/grpc/interceptor.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+// UnaryClientErrorInterceptor transfers an error to status error in unary client
+func UnaryClientErrorInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn,
+	invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	err := invoker(ctx, method, req, reply, cc, opts...)
+	return toServiceError(err)
+}
+
+// toStatusError converts ServiceError to gRPC status error
+func toStatusError(err error) error {
```
[the remark this answers is not in the document]
Comment (sysvm) [reply util/grpc/interceptor.go:43]: This function is used to transfer status error to service error and should use error interface.

File: model/errors/rpc_error.go [hunk 5]
```diff
@@ -120,10 +122,10 @@ var (
 func InnerErrorToGRPCError(err error) error {
 	if errors.Is(err, gorm.ErrRecordNotFound) ||
 		errors.Is(err, ErrNoSuchObject) {
-		return status.Errorf(codes.NotFound, "Object is not found")
+		return errorstypes.Error(NoSuchObjectErrCode, "object is not found")
 	}
 	if errors.Is(err, ErrCheckQuotaEnough) {
-		return status.Errorf(codes.PermissionDenied, "Quota is not enough")
+		return status.Errorf(codes.PermissionDenied, "quota is not enough")
```
Comment (RenRick) [inline model/errors/rpc_error.go:128]: Should be "DBQuotaNotEnoughErrCode"?

File: util/grpc/interceptor.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		if e != nil {
+			return st.Err()
+		}
+		return ds.Err()
+	}
+	return err
+}
+
+// toServiceError converts gRPC status error to ServiceError
+func toServiceError(err error) error {
```
[the remark this answers is not in the document]
Comment (sysvm) [reply util/grpc/interceptor.go:63]: ditto

File: model/errors/rpc_error.go (hunk 5 above)
Comment (sysvm) [reply model/errors/rpc_error.go:128]: yes, this function will be removed.

**Pull request closed without merging**